examples/todo/modules/data_l1/src/main/scala/com/my/data_l1/DataL1Service.scala

```diff
@@ -0,0 +1,113 @@
+package com.my.data_l1
+
+import cats.data.NonEmptyList
+import cats.effect.Async
+import cats.syntax.all._
+
+import org.tessellation.currency.dataApplication._
+import org.tessellation.currency.dataApplication.dataApplication.{
+  DataApplicationBlock,
+  DataApplicationValidationErrorOr
+}
+import org.tessellation.json.JsonSerializer
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.security.Hasher
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.signature.Signed._
+
+import com.my.data_l1.DataL1NodeContext.syntax.DataL1NodeContextOps
+import com.my.shared_data.lib.CirceOps.implicits._
+import com.my.shared_data.schema.Updates.TodoUpdate
+import com.my.shared_data.schema.{CalculatedState, OnChain}
+
+import io.circe.{Decoder, Encoder}
+import org.http4s._
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+
+object DataL1Service {
+
+  def make[F[+_]: Async: JsonSerializer: Hasher]: F[BaseDataApplicationL1Service[F]] =
+    for {
+      validator <- Async[F].pure(DataL1Validator.make[F])
+      dataApplicationL1Service = makeBaseApplicationL1Service(validator)
+    } yield dataApplicationL1Service
+
+  private def makeBaseApplicationL1Service[F[+_]: Async: JsonSerializer](
+    validator: DataL1Validator[F, TodoUpdate, OnChain]
+  ): BaseDataApplicationL1Service[F] =
+    BaseDataApplicationL1Service[F, TodoUpdate, OnChain, CalculatedState](
+      new DataApplicationL1Service[F, TodoUpdate, OnChain, CalculatedState] {
+
+        override def serializeState(state: OnChain): F[Array[Byte]] =
```

Comment:
do you override something or you provide an implementation? I think you override nothing so the `override` modifier is unnecessary.

Reply:
I'm not sure I understand. This does override the `serializeState` method from `BaseDataApplicationService` which the `BaseDataApplicationL1Service` [inherits from](https://github.com/Constellation-Labs/tessellation/blob/36e8a0fa551842000823584fad9c951474c065dc/modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala#L110).

Reply:
as far as I see, `BaseDataApplicationService` does not implement any of these methods.
![image](https://github.com/user-attachments/assets/9f0dafc0-f816-4432-9751-c496702c4758)

so you don't override the default implementation, but you provide the first implementation there. hence, you "override" nothing.

Reply:
Oh, I see. You have implemented the `data l1` here only. I think you should provide a single `DataApplicationService` and reuse it for each layer 🤔 you should not implement each layer independently 🤔

Reply:
oh, I see, I believe I misunderstood your comment at first. You are correct that there are no other implementations that are being overridden so it is not strictly needed in this case. I was under the impression that it was common practice to use the override modifier to signify that an implementation is required due to inheritance.